sdk/core/azure-core/src/base64.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/core/base64.hpp"
+
+#include <azure/core/platform.hpp>
+
+#if defined(AZ_PLATFORM_WINDOWS)
+#include <windows.h>
+#elif defined(AZ_PLATFORM_POSIX)
+#include <openssl/bio.h>
+#include <openssl/buffer.h>
+#include <openssl/evp.h>
+#endif
+
+#include <string>
+#include <vector>
+
+using namespace Azure::Core;
+
+#if defined(AZ_PLATFORM_WINDOWS)
```

Comment:
I would move this to inside the function to only what it is different between win and lin instead of duplicating the entire method.
Makes it easier to maintain

Reply:
Other than the function signature, the entire implementation is different and moving it into the function means I'd need to add the ifdef in two places (since we have two functions). I don't see much of a win here.

> Makes it easier to maintain

Do you still see some benefit in this case?

Reply:
well, I see the benefit of going to a single place for updating and then finding that there are more than one implementations.  In the other hand, some IDE might take you to one implementation without you noticing there's another as well. You need to be curious to check if there might be more.

I would push harder if this would be in the headers, but since it is cpp, I think is fine. The benefit is not super important at the end.